src/main/java/org/openstreetmap/atlas/checks/validation/areas/ShadowDetectionCheck.java

```diff
@@ -250,60 +251,33 @@ private boolean neighborsHeightContains(final AtlasObject part, final AtlasObjec
         try
         {
             // Set partMinHeight
-            final double partMinHeight = partTags.containsKey(MinHeightTag.KEY)
-                    ? Double.parseDouble(partTags.get(MinHeightTag.KEY))
-                    : Double.parseDouble(partTags.get(BuildingMinLevelTag.KEY))
-                            * LEVEL_TO_METERS_CONVERSION;
+            final double partMinHeight = MinHeightTag.get(part).map(Altitude::asMeters)
+                    .orElseGet(() -> Double.parseDouble(partTags.get(BuildingMinLevelTag.KEY))
+                            * LEVEL_TO_METERS_CONVERSION);
+
             // Set partMaxHeight
-            final double partMaxHeight;
-            if (partTags.containsKey(HeightTag.KEY))
-            {
-                partMaxHeight = Double.parseDouble(partTags.get(HeightTag.KEY));
-            }
-            else if (partTags.containsKey(BuildingLevelsTag.KEY))
-            {
-                partMaxHeight = Double.parseDouble(partTags.get(BuildingLevelsTag.KEY))
-                        * LEVEL_TO_METERS_CONVERSION;
-            }
-            else
-            {
-                // Default to 0 height above the minimum
-                partMaxHeight = partMinHeight;
-            }
+            final double partMaxHeight = HeightTag.get(part).map(Altitude::asMeters)
```

Comment:
not specifically this line but does this check if the part max height is below the min height?

Reply:
It does. At the end of this method there is an attempt to create a range from the 2 height values. If the min is above the max an error is thrown, and the method returns false. This would cause the broken part to appear floating to the check, and it would be flagged.

Reply:
dope